modules/rubiconBidAdapter.js

```diff
@@ -1151,6 +1137,104 @@ function partitionArray(array, size) {
   return array.map((e, i) => (i % size === 0) ? array.slice(i, i + size) : null).filter((e) => e)
 }
 
+/**
+ * add userId to data
+ * @param {string} source is used to get the matching eid array item
+ * @param {UserIdEid[]} userIdEids eid array from the bid request object
+ * @param {Object} data sets the object that the user id properties will be added to
+ * @param {Array} paramMappings
+ */
+export function setUserIdParam(source, userIdEids, data, paramMappings) {
+  const userIdEid = find(userIdEids, eid => eid.source === source);
+  if (userIdEid) {
+    paramMappings.forEach(paramMapping => {
+      const paramName = Object.keys(paramMapping)[0];
+      data[paramName] = paramMapping[paramName].map(paramPath => {
+        const paramValue = utils.deepAccess(userIdEid, paramPath);
+        return Array.isArray(paramValue) ? paramValue.join(',') : paramValue;
+      }).join('^');
+    });
+  }
+}
+
+/**
+ * @param {Object} bidRequest
+ * @param {Object} data
+ */
+export function createUserIdParams(bidRequest, data) {
+  if (bidRequest.userIdAsEids && bidRequest.userIdAsEids.length) {
+    const PROP_ID = 'uids.0.id';
+    const PROP_ATYPE = 'uids.0.atype';
+    const DEFAULT_PROPS = [PROP_ID, PROP_ATYPE];
+
+    [
+      // britepoolId
```

Comment:
We should not be listing out all each ID. There will be more and the goal is to not have to update this code to pass the rest through. Please review the algorithm defined in internal ticket HB-10116.